code/modules/maps/ruins.dm

```diff
@@ -36,6 +36,17 @@ GLOBAL_LIST_EMPTY(banned_ruin_ids)
 			handled_ruin_paths += ruin.type
 			continue
 
+		if((ruin.template_flags & TEMPLATE_FLAG_PORT_SPAWN) && (ruin.spawns_in_current_sector()))
```

Comment:
The check below already handles the spawns_in_current_sector(), what is the point of processing it here too, over moving this block below it and just cycle the potential ruins that it already filtered?